src/logic/generic_save_handler.cc

```diff
@@ -124,6 +124,7 @@ void GenericSaveHandler::save_file() {
 }
 
 GenericSaveHandler::Error GenericSaveHandler::save() {
+	MutexLock m(MutexLock::ID::kLua);
```

Comment:
is this really related to this PR and its described content

Reply:
yes, this is to fix a testsuite deadlock that only happens with this change